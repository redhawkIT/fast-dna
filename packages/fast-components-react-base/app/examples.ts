import ButtonExamples from "../src/button/examples.data";
export { ButtonExamples };
