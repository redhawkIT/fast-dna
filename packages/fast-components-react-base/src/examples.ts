import * as React from "react";

export type IExampleData<T> = T;

export interface IGenericExample<T> {
    name: string;

    component: React.ComponentClass;

    data: Array<IExampleData<T>>;
}
