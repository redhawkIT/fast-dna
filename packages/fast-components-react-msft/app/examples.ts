import ButtonExample from "../src/button/examples.data";
export {ButtonExample};

import DividerExample from "../src/divider/examples.data";
export {DividerExample};

import ToggleExamples from "../src/toggle/examples.data";
export {ToggleExamples};
