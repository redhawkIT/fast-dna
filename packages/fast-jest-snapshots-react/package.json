{
  "name": "@microsoft/fast-jest-snapshots-react",
  "description": "A small utility for automating Jest snapshot creation of React components",
  "version": "1.6.0",
  "author": {
    "name": "Microsoft",
    "email": "fwq@microsoft.com"
  },
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/Microsoft/fast-dna.git"
  },
  "private": true,
  "main": "./dist/index.js",
  "types": "./dist/index.d.ts",
  "scripts": {
    "prepare": "tsc -p ./tsconfig.json",
    "test": "npm run tslint",
    "start": "webpack-dev-server",
    "tslint": "tslint -c ./tslint.json '**/*.ts{,x}'",
    "tslint:fix": "tslint -c ./tslint.json --fix '**/*.ts{,x}'"
  },
  "devDependencies": {
    "@microsoft/fast-tslint-rules": "^1.6.0",
    "@types/jest": "^22.2.2",
    "jest": "^22.4.3",
    "react": "^16.3.0",
    "ts-jest": "^22.4.2",
    "tslint": "^5.9.1",
    "typescript": "^2.8.1"
  },
  "dependencies": {
    "@types/react": "^16.3.0",
    "react-test-renderer": "^16.3.0"
  },
  "peerDependencies": {
    "jest": "^22.0.0",
    "react": "^16.3.0"
  }
}
