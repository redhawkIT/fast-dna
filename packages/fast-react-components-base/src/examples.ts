import * as React from "react";

export type IExampleData<T> = T;

export interface IGenericExample<T> {
    name: string;

    component: React.ComponentClass;

    data: IExampleData<T>[];
}