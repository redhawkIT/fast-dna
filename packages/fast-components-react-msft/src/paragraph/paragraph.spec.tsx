import * as React from "react";
import * as Adapter from "enzyme-adapter-react-16";
import { configure, shallow } from "enzyme";
import examples from "./examples.data";
import { generateSnapshots } from "@microsoft/fast-jest-snapshots-react";
import MSFTParagraph, {
    IParagraphHandledProps,
    IParagraphManagedClasses,
    IParagraphUnhandledProps,
    ParagraphLevel,
    ParagraphProps,
} from "./paragraph";
import { Paragraph } from "./index";

/*
 * Configure Enzyme
 */
configure({adapter: new Adapter()});

describe("paragraph snapshots", (): void => {
    generateSnapshots(examples);
});

describe("paragraph", (): void => {
    test("should have a displayName that matches the component name", () => {
        expect((MSFTParagraph as any).name).toBe(MSFTParagraph.displayName);
    });

    test("should return an object that includes all valid props which are not enumerated as handledProps", () => {
        const handledProps: IParagraphHandledProps = {
            size: ParagraphLevel._1
        };

        const unhandledProps: IParagraphUnhandledProps = {
            "aria-hidden": true
        };

        const props: IParagraphHandledProps & IParagraphUnhandledProps = {...handledProps, ...unhandledProps};

        const rendered: any = shallow(
            <Paragraph {...props} />
        );

        const paragraph: any = rendered.first().shallow();

        expect(paragraph.prop("aria-hidden")).toEqual(true);
    });

    test("should render the correct `size` when `size` prop is passed", () => {
        const rendered: any = shallow(
            <Paragraph size={ParagraphLevel._2} />
        );
        const paragraph: any = rendered.first().shallow();

        expect(paragraph.instance().props.size).toEqual(ParagraphLevel._2);
    });

    test("should render a default `size` of `ParagraphLevel._3` if no `size` prop is passed", () => {
        const rendered: any = shallow(
            <Paragraph />
        );
        const paragraph: any = rendered.first().shallow();

        expect(paragraph.instance().props.size).toEqual(ParagraphLevel._3);
    });
});
