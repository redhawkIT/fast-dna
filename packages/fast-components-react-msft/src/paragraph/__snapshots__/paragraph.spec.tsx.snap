// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`paragraph snapshots Paragraph: 0 1`] = `
<p
  className="typography-0-1-1 typography__7-0-1-8 paragraph-0-1-11 paragraph-0-1-11 paragraph__3-0-1-14"
>
  Paragraph test string
</p>
`;

exports[`paragraph snapshots Paragraph: 1 1`] = `
<p
  className="typography-0-1-1 typography__5-0-1-6 paragraph-0-1-11 paragraph-0-1-11 paragraph__1-0-1-12"
  data-sketch-symbol="Paragraph 1"
>
  Paragraph test string
</p>
`;

exports[`paragraph snapshots Paragraph: 2 1`] = `
<p
  className="typography-0-1-1 typography__6-0-1-7 paragraph-0-1-11 paragraph-0-1-11 paragraph__2-0-1-13"
  data-sketch-symbol="Paragraph 2"
>
  Paragraph test string
</p>
`;

exports[`paragraph snapshots Paragraph: 3 1`] = `
<p
  className="typography-0-1-1 typography__7-0-1-8 paragraph-0-1-11 paragraph-0-1-11 paragraph__3-0-1-14"
  data-sketch-symbol="Paragraph 3"
>
  Paragraph test string
</p>
`;
